### DISCUSSION TITLE
\ref autocomplete

### DISCUSSION THREAD

**Discussion by diogosfs** [James-Yu/LaTeX-Workshop#3573] [Q&A]
Seems like `\ref` autocomplete needs to refresh the file?

![example0](https://user-images.githubusercontent.com/15656068/205994533-8056dba8-9a40-42b5-9cc6-91538e49b7dd.gif)

Also I have instances where `\ref` autocomplete won't work at all for larger projects where I have chapters in individual files. It would be nice to have `ref` work across the project and also at least within the same file.

Updates:
- Update on this discussion is that once I have compiled my latex project, the references seem to appear and now references are added to the pool of references on autocomplete. Not sure if that is the desired behavior.

**Comment by CharlieDerby** [comment]
I have also noticed this issue, it would be better if the pool of references would auto-update without needing to build the file for each new label. Is this possible?

**Comment by James-Yu** [comment]
`latex-workshop.intellisense.update.aggressive.enabled`

**Reply by CharlieDerby** [reply]
Thanks! Also thanks for putting the extension together :)

**Reply by CharlieDerby** [reply]
Hi, I actually just tried this and it doesn't seem to have worked, even if I reload the window after applying the setting. Here's a video of what I'm seeing: 

https://github.com/James-Yu/LaTeX-Workshop/assets/33317223/c97f14d7-e6b9-4fcc-845c-4f4a5f456906

Summary of video:
- settings.json has `"latex-workshop.intellisense.update.aggressive.enabled": true` entry
- I add a label
- I type `\ref{` and I don't get a suggestion of the label I just added
- I build the file
- I get the suggestion as expected

Reloading the Window via `Ctrl+Shift+P` does not change this behaviour. If the most recent save of the file doesn't have a reference then it won't get suggested.

**Reply by James-Yu** [reply]
It is caused by a bug, fixed in 2d301ce869893323e0bf427b3dbfe11bfd584427

**Reply by CharlieDerby** [reply]
Looks like it works now, thanks!
